src/dmd/e2ir.d

```diff
@@ -6323,17 +6334,17 @@ elem *callCAssert(IRState *irs, const ref Loc loc, Expression exp, Expression em
     if (irs.params.isOSX)
     {
         // __assert_rtn(func, file, line, msg);
-        const(char)* id = "";
-        FuncDeclaration fd = irs.getFunc();
-        if (fd)
-            id = fd.toPrettyChars();
-        const len = strlen(id);
-        Symbol *si = toStringSymbol(id, len, 1);
-        elem *efunc = el_ptr(si);
-
+        elem* efunc = getFuncName();
         auto eassert = el_var(getRtlsym(RTLSYM_C__ASSERT_RTN));
         ea = el_bin(OPcall, TYvoid, eassert, el_params(elmsg, eline, efilename, efunc, null));
     }
+    else version (CRuntime_Musl)
```

Comment:
```
if(...) {...}
else version(...) {...}
else {...}
```
looks _really_ weird...

Reply:
It does, but it actually is quite natural when you think of it. After all:
```D
if (a) {}
else
{
    if (b) {}
    else {}
}
```
is equivalent to:
```D
if (a) {}
else if (b) {}
else {}
```

And I suppose the following would look less weird:
```D
if (a) {}
else
{
    version (b) {}
    else {}
}
```

but works just the same.